pyqtgraph/exporters/CSVExporter.py

```diff
@@ -36,23 +34,48 @@ def export(self, fileName=None):
         data = []
         header = []
 
-        appendAllX = self.params['columnMode'] == '(x,y) per plot'
+        appendAllX = self.params['columnMode'] == '(x,y,err_y,err_x) per plot'
 
         for i, c in enumerate(self.item.curves):
             cd = c.getData()
             if cd[0] is None:
                 continue
             data.append(cd)
+
             if hasattr(c, 'implements') and c.implements('plotData') and c.name() is not None:
                 name = c.name().replace('"', '""') + '_'
                 xName, yName = '"'+name+'x"', '"'+name+'y"'
             else:
                 xName = 'x%04d' % i
                 yName = 'y%04d' % i
+                e_yName = 'e_y%04d' % i
+                e_xName = 'e_x%04d' % i
+
             if appendAllX or i == 0:
-                header.extend([xName, yName])
+                header.extend([xName, yName,e_yName,e_xName ])
             else:
-                header.extend([yName])
+                header.extend([yName,e_yName,e_xName ])
+
+        for indx, item in enumerate(self.item.items):
+
+            if self.item.items[indx].__class__.__name__ == "InfiniteLine":                   
+                continue   
+            if self.item.items[indx].__class__.__name__ == "FillBetweenItem":                               
+                continue
+            if self.item.items[indx].__class__.__name__ == "TextItem":
+                continue
+
+            if "top" in self.item.items[indx].opts:
+                yerr=self.item.items[indx].opts["top"]
+                data[1] = list(data[1])
+                data[1].append(yerr)
+                data[1] = tuple(data[1])
+
+            if "left" in self.item.items[indx].opts:
```

Comment:
these if-statements have some near identical code, consider consolidating them.  

One suggestion would be to use a for loop

```python

for option in ['top', 'left']:
    if option in item.opts:
        err = item.opt[option]
        data[1] = list(data[1]
        data[1].append(err)
        data[1] = tuple(data[1])
```